src/ml/neural_net/image_augmentation.cpp

```diff
@@ -117,5 +117,98 @@ image_augmenter::result resize_only_image_augmenter::prepare_images(
   return res;
 }
 
+image_augmenter::result processed_image_augmenter::prepare_images(
+    std::vector<labeled_image> source_batch) {
+  result res;
+
+  const size_t n = opts_.batch_size;
+  constexpr size_t c = 3;
+
+  std::vector<turi::neural_net::shared_float_array> images_to_aug;
+  std::vector<turi::neural_net::shared_float_array> ann_batch;
+  std::vector<turi::neural_net::shared_float_array> pred_batch;
+
+  // Decode a batch of images to raw format
+  for (size_t i = 0; i < n; i++) {
+    size_t input_height = source_batch[i].image.m_height;
+    size_t input_width = source_batch[i].image.m_width;
+    std::vector<float> img(input_height * input_width * c, 0.f);
+    // unsigned char *outptr = reinterpret_cast<unsigned char *>(img.data());
+    image_util::copy_image_to_memory(source_batch[i].image, img.data(),
+                                     {input_width * c, c, 1},
+                                     {input_height, input_width, c}, true);
+
+    std::transform(img.begin(), img.end(), img.begin(),
+                   [](float pixel) -> float { return pixel / 255; });
+    shared_float_array image_to_aug =
+        shared_float_array::wrap(img, {input_height, input_width, c});
+    images_to_aug.push_back(image_to_aug);
+    std::vector<float> annotation(source_batch[i].annotations.size() * 6);
+    size_t x = 0;
+    for (size_t j = 0; j < source_batch[i].annotations.size(); j++) {
+      annotation[x] = source_batch[i].annotations[j].identifier;
+      annotation[x + 1] = source_batch[i].annotations[j].bounding_box.x;
+      annotation[x + 2] = source_batch[i].annotations[j].bounding_box.y;
+      annotation[x + 3] = source_batch[i].annotations[j].bounding_box.height;
+      annotation[x + 4] = source_batch[i].annotations[j].bounding_box.width;
+      annotation[x + 5] = source_batch[i].annotations[j].confidence;
+      x = (j + 1) * 6;
```

Comment:
Instead of defining `x` outside the loop and updating it at the end of each iteration, could you just define it at the beginning of each iteration, as `size_t x = j * 6`? Let's also use a more informative name, something like `annotation_offset`.